pkg/apis/application/v1alpha1/types.go

```diff
@@ -166,14 +165,33 @@ func (h *ApplicationSourceHelm) IsZero() bool {
 	return h == nil || (h.ReleaseName == "") && len(h.ValueFiles) == 0 && len(h.Parameters) == 0
 }
 
+type KustomizeImage string
+
+// if the image name matches (i.e. up to the first "=" symbol)
+func (i KustomizeImage) Match(j KustomizeImage) bool {
+	return strings.HasPrefix(string(i), strings.Split(string(j), "=")[0])
```

Comment:
This does not catch all cases. E.g. it won't work if user tries to override image tag:

```
argocd app set argo-cd -p argoproj/argocd:v1.1.3
argocd app set argo-cd -p argoproj/argocd:v1.1.4
```

results into 
```
  kustomize:
    images:
      - 'argoproj/argocd:v1.1.3'
      - 'argoproj/argocd:v1.1.4'
```
instead of 
```
  kustomize:
    images:
      - 'argoproj/argocd:v1.1.4'
```

Reply:
good spot - needs fixing